aws_ec2_assign_elastic_ip/__init__.py

```diff
@@ -63,6 +67,14 @@ def main():
     if not address:
         sys.exit(1)
 
+    # Unassign existing IP if needed
+    if associated_address and args.replace:
+        if args.dry_run:
+            logger.info('Would unassign IP {0}'.format(
+                associated_address.public_ip))
+        else:
+            _unassign_address(associated_address)
```

Comment:
There is a race condition here where an instance can throw away its existing IP but get beaten to the "vip-ip" by another instance. This isn't a major problem since it fixes itself next time the commands run.